src/background.js

```diff
@@ -116,6 +117,23 @@ ipcMain.on('shutdown-finished', () => {
   app.quit()
 })
 
+ipcMain.on('app_version', event => {
+  event.sender.send('app_version', { version: app.getVersion() })
+})
+
+// Ipc event received from the client to restart Sheikah and install new version
+ipcMain.on('restart_app', () => {
+  win.webContents.send('log', 'restart app')
+  setImmediate(() => {
+    app.removeAllListeners('close')
+    forceQuit = true
+    if (win != null) {
+      win.close()
+    }
+    autoUpdater.quitAndInstall(false)
+  })
+})
```

Comment:
Why are you using `setImmediate`? AFAIK, is not a standard and its use is discouraged.
On the other hand, why do you emit a `restart_app` event to the client to receive that event again in the background? Could you simplify the flow calling directly this in the background?
```suggestion
ipcMain.on('restart_app', () => {
  win.webContents.send('log', 'restart app')
    app.removeAllListeners('close')
    forceQuit = true
    if (win != null) {
      win.close()
    }
    autoUpdater.quitAndInstall(false)
})
```